node/src/components/contract_runtime/operations.rs

```diff
@@ -67,6 +69,12 @@ pub fn execute_finalized_block(
         });
     }
 
+    if executable_block.era_report.is_some() && next_era_gas_price.is_none() {
+        return Err(BlockExecutionError::FailedToGetNewEraGasPrice {
+            era_id: executable_block.era_id.successor(),
+        });
+    }
```

Comment:
this is basically fatal, isn't it?

Reply:
Yes, but it gets translated into a fatal by the calling logic